helix-core/src/path.rs

```diff
@@ -1,125 +1,16 @@
-use etcetera::home_dir;
-use std::path::{Component, Path, PathBuf};
+use helix_stdx::path::{canonicalize, fold_home_dir, normalize};
```

Comment:
IMO path hbdeling doesn't belong on helix-core at all. I would like helix-core to just be about "core editing/editor components".

I see that you introduce this to handle the dependency on the CWD handling in helix-loader.

I think there are two alternatives here:
* move the cwd handling to stdx::path. While its somewhat in spirit of helix-loader I think ultimately what we are doing is essentially wrapping `std::env::current_dir` to make preventing it from crashing.
* move this Module to hwlix-loader

I would prefer the first variant since it really just feels like the cwd stuff is making std more robust rather than being any actual loading.